src/Types/BigIntType.php

```diff
@@ -28,12 +37,25 @@ public function getBindingType(): ParameterType
     /**
      * @param T $value
      *
-     * @return (T is null ? null : string)
+     * @return (T is null ? null : int|string)
      *
      * @template T
      */
-    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): ?string
+    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int|string|null
     {
-        return $value === null ? null : (string) $value;
+        if ($value === null || is_int($value)) {
+            return $value;
+        }
+
+        if ($value > PHP_INT_MIN && $value < PHP_INT_MAX) {
```

Comment:
Shouldn't this be `>= && <=`?

Reply:
no

Reply:
see https://3v4l.org/I9jI9 repro - may I know why the valid int extremes should not be casted?

Reply:
Because I cannot detect them safely.

Reply:
will `$v === (string) (int) $v` work?

Reply:
@derrabus I really feel this is not matching the expected behaviour and documentation.

Reply:
It is your PR and I am just reporting an issue to you privately. The docs is clear the whole representible range is supported.